### ISSUE TITLE
wrong type used for calculations when the target is of the correct type

### ISSUE THREAD

**Issue by xmonader** [vlang/v#10827]
![image](https://user-images.githubusercontent.com/64129/125927799-b6f53511-8dc0-4f02-96c5-df2732ecb848.png)

apparently I need on the right hand side to cast the first item to u64 

![image](https://user-images.githubusercontent.com/64129/125927848-905dfe60-7601-4232-ab0c-acdc81fbf29b.png)

**Comment by JalonSolov** [comment]
This is expected behavior.  The default numeric type is `int`.  If you want to use a different type, you have to tell V what you want to use.

**Comment by xmonader** [comment]
> This is expected behavior. The default numeric type is `int`. If you want to use a different type, you have to tell V what you want to use.

How?

**Comment by JalonSolov** [comment]
Exactly as you did - by casting the first number to `u64` type.

**Comment by xmonader** [comment]
@JalonSolov the target type is already known, otherwise I'd have to mismatch with the backend types and if that type is defined in a nested structure it won't be as simple as the workaround I provided in the example.

**Comment by JalonSolov** [comment]
Your example is not V.  V doesn't use that syntax.

Especially in the REPL, with no other reference, all V sees is `int` values unless you tell it otherwise.  There is no way for it to know "Oh, I meant it to be a u64."  So it is simply doing integer math and the value overflows, and you get the large negative number instead of the even larger positive number you wanted.

By casting the first number to `u64`, you tell V that you want it to do everything with `u64`, so you get the number you expected.

**Comment by xmonader** [comment]
![image](https://user-images.githubusercontent.com/64129/125970407-6112a13b-41fd-42c6-b16c-ad38acb9a5de.png)
this example is V in `v repl` where I defined a variable as u64 and initialized it to 0 then tried to assign it the value of `1024*1024*1024*10`


> By casting the first number to u64, you tell V that you want it to do everything with u64, so you get the number you expected.

that's what _should V infer from the type of variable `y` being a `u64`_, that `y` is as u64 and wants to do everything with that.  also maybe it's related to these kinds of issues as well #10737 (where the type is overlooked)

**Comment by JalonSolov** [comment]
It's still not correct.  V does the calculation on the right first, with int values, then stores the result in the `u64` variable you created.  There is no inference implied.  If you do a calculation with ints that overflows ints, you get unexpected results.  It can just display that overflowed value in a different way because the thing it was stored in was large enough to show a larger positive value.

If you want it to do the calculations as u64 values, you either have to cast the first number to tell it, or you have to do the calculations on the u64 variable.

For example, this shows what you expected:
```v
mut y := u64(1024)

y *= 1024*1024*10

println(y)
```
while this is equivalent to what you had in the REPL:
```v
mut y := u64(1024*1024*1024*10)

println(y)
```
In other words - it does the calculation with ints, then casts the RESULT to u64, but it doesn't do the calculation AS u64 values.
